Sources/MuxUploadSDK/Internal Utilities/Reporter.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  Reporter.swift
+//  
+//
+//  Created by Liam Lindner on 3/16/23.
+//
+
+import Foundation
+
+class Reporter: NSObject {
+    var session: URLSession?
+    var pendingUploadEvent: UploadEvent?
```

Comment:
Will there ever be a need to keep around >1 `UploadEvent`?

Reply:
I had the same thought when I wrote this 😅

It would be better to queue up events rather than keeping this one variable around between the redirect. We might want to build this out a little further and POST events in batches, but that's probably not as important on this first iteration.